segmentation/utils/spatial_analysis.py

```diff
@@ -0,0 +1,245 @@
+import pandas as pd
+import numpy as np
+import scipy
+import statsmodels
+from statsmodels.stats.multitest import multipletests
+
+# Erin's Data Inputs
+
+# cell_array = pd.read_csv("/Users/jaiveersingh/Downloads/SpatialEn"
+#                          "richment/granA_cellpheno_CS-asinh-norm_revised.csv")
+# marker_thresholds = pd.read_csv("/Users/jaiveersingh/Downloads/Sp"
+#                                 "atialEnrichment/markerThresholds.csv")
+# dist_matrix = np.asarray(pd.read_csv("/Users/jaiveersingh/Documen"
+#                                      "ts/MATLAB/distancesMat5.csv",
+#                                      header=None))
+
+
+def compute_close_cell_num(patient_data, patient_data_markers, thresh_vec,
+                           dist_mat, marker_num, dist_lim, cell_label_idx):
+    """Finds positive cell labels and creates matrix with counts for cells positive for corresponding markers.
+
+    Args:
+        patient_data: cell expression data for the specific point
+        patient_data_markers: cell expression data of markers
+            for the specific point
+        thresh_vec: matrix of thresholds column for markers
+        dist_mat: cells x cells matrix with the euclidian
+            distance between centers of corresponding cells
+        marker_num: number of markers in expresion data
+        dist_lim: threshold for spatial enrichment distance proximity
+        cell_label_idx: column with cell labels
+
+    Returns:
+        close_num: marker x marker matrix with counts for cells
+            positive for corresponding markers
+        marker1_num: list of number of cell labels for marker 1
+        marker2_num: list of number of cell labels for marker 2"""
+
+    # Create close_num, marker1_num, and marker2_num
+    close_num = np.zeros((marker_num, marker_num), dtype='int')
+    marker1_num = []
+    marker2_num = []
+
+    for j in range(0, marker_num):
+        # Identify cell labels that are positive for respective markers
+        marker1_thresh = thresh_vec.iloc[j]
+        marker1posinds = patient_data_markers[patient_data_markers.columns[j]] > marker1_thresh
+        marker1poslabels = patient_data.loc[marker1posinds, patient_data.columns[cell_label_idx]]
+        marker1_num.append(len(marker1poslabels))
+        for k in range(0, marker_num):
+            # Identify cell labels that are positive for above marker and all other markers
```

Comment:
Here you're describing what the for loop does across the entire loop (all other markers). What what you want to describe is what it is doing inside a single iteration of the loop, which is what someone reading the code will be trying to figure out. 


I would change this to 'identify all cells that are positive for kth marker'